14969 vuln software seeder

mostlikelee opened the pull request:
# Checklist for submitter
#14969 

If some of the following don't apply, delete the relevant line.

<!-- Note that API documentation changes are now addressed by the product design team. -->

- [ ] Changes file added for user-visible changes in `changes/` or `orbit/changes/`.
  See [Changes files](https://fleetdm.com/docs/contributing/committing-changes#changes-files) for more information.
- [ ] Documented any permissions changes (docs/Using Fleet/manage-access.md)
- [ ] Input data is properly validated, `SELECT *` is avoided, SQL injection is prevented (using placeholders for values in statements)
- [ ] Added support on fleet's osquery simulator `cmd/osquery-perf` for new osquery data ingestion features.
- [ ] Added/updated tests
- [ ] Manual QA for all new/changed functionality
  - For Orbit and Fleet Desktop changes:
    - [ ] Manual QA must be performed in the three main OSs, macOS, Windows and Linux.
    - [ ] Auto-update manual QA, from released version of component to new version (see [tools/tuf/test](../tools/tuf/test/README.md)).

getvictor commented on tools/seed_data/seed_vuln_data.go:
```diff
@@ -0,0 +1,351 @@
+	Source           string `db:"source"`
+	BundleIdentifier string `db:"bundle_identifier"`
+	Release          string `db:"release"`
+	VendorOld        string `db:"vendor_old"`
```
Do we need `VendorOld` here?

getvictor commented on tools/seed_data/seed_vuln_data.go:
```diff
@@ -0,0 +1,351 @@
+	defer file.Close()
+
+	reader := csv.NewReader(file)
+	reader.Comma = ',' // Set the delimiter to comma
```
comma is the default. I don't think this line is needed.

getvictor commented on tools/seed_data/seed_vuln_data.go:
```diff
@@ -0,0 +1,351 @@
+		return nil, err
+	}
+
+	var softwares []Software
```
The plural of software is software

getvictor commented on tools/seed_data/seed_vuln_data.go:
```diff
@@ -0,0 +1,351 @@
+	var existingID int64
+	query := `SELECT id FROM software WHERE name = ? AND version = ? AND source = ?`
+	err := db.Get(&existingID, query, software.Name, software.Version, software.Source)
+	if err != nil && err != sql.ErrNoRows {
```
```suggestion
	if err != nil && errors.Is(err, sql.ErrNoRows) {
```
I think `errors.Is` should generally be used, since it can also handle wrapped errors.

getvictor commented on tools/seed_data/seed_vuln_data.go:
```diff
@@ -0,0 +1,351 @@
+		log.Fatal(err)
+	}
+
+	// Insert macos software
```
```suggestion
	// Insert macOS software
```

getvictor reviewed (changes requested):
Can you add a README.md to explain usage?

mostlikelee commented:
Can't seem to replicate here.  Tried different combinations of `dpkg` and `apt` installations.  Possible that this self resolved in v1.18? 

```
tim@tim-ubuntu:~$ sudo apt install ./fleet-osquery_1.18.3_amd64.deb
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Note, selecting 'fleet-osquery' instead of './fleet-osquery_1.18.3_amd64.deb'
fleet-osquery is already the newest version (1.18.3).
0 upgraded, 0 newly installed, 0 to remove and 47 not upgraded.
tim@tim-ubuntu:~$ sudo dpkg -i fleet-osquery_1.18.3_amd64.deb
(Reading database ... 181819 files and directories currently installed.)
Preparing to unpack fleet-osquery_1.18.3_amd64.deb ...
Removed /etc/systemd/system/multi-user.target.wants/orbit.service.
Unpacking fleet-osquery (1.18.3) over (1.18.3) ...
Setting up fleet-osquery (1.18.3) ...
Created symlink /etc/systemd/system/multi-user.target.wants/orbit.service → /lib/systemd/system/orbit.service.
tim@tim-ubuntu:~$ sudo dpkg -i fleet-osquery_1.18.3_amd64.deb
(Reading database ... 181819 files and directories currently installed.)
Preparing to unpack fleet-osquery_1.18.3_amd64.deb ...
Removed /etc/systemd/system/multi-user.target.wants/orbit.service.
Unpacking fleet-osquery (1.18.3) over (1.18.3) ...
Setting up fleet-osquery (1.18.3) ...
Created symlink /etc/systemd/system/multi-user.target.wants/orbit.service → /lib/systemd/system/orbit.service.
```

getvictor commented:
@mostlikelee What about adding a README.md to explain usage?
